### ISSUE TITLE
Piezo Projector

### ISSUE THREAD

**Issue by NrkSed** [Anuken/Mindustry-Suggestions#3347]
### Describe the content or mechanics you are proposing.

Piezo Projector will be a block unlocked after the Force projector (after the player has researched Surge Alloy) 
It will cast a shield which is the same size and has the same heath as the Force projector's shield, and will mostly act in the same way except for having a lighter/darker shade of shield.
Building it will cost an additional 30 Phase fabric and 45 Surge alloy

The key difference is that when the shield is damaged it will output a power dependant on the amount of damage. 

For instance:
Dagger Bullet: 1.5 power units/bullet
Atrax Slag orb: 4 power units/orb
Vela laser: 350 power units/s
Reign shell: 60-80 power units/bullet

Nothing different happens if it is cooled with water, however when cooled with Cyofluid, power production is 1.5x as well as the shield being cooled

### Describe how you think this content will improve the game. If you're proposing new content, mention how it may add more gameplay options or how it will fill a new niche.

I dont know whether this will be OP or useless, and the stats and power:damage ratios are very vague, i just wanted to show the general idea

This block will come in handy in Late game where there are a huge amount of enemies and your weapons are drawing alot of power when they approach, causing a large power dip.

### Before making this issue, check the boxes below to confirm that you have acknowledged them.

- [X] I have checked the [Trello](https://trello.com/b/aE2tcUwF/mindustry-trello) to make sure my suggestion isn't planned or implemented in a development version.
- [X] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet.
- [X] I have read [README.md](https://github.com/Anuken/Mindustry-Suggestions/blob/master/README.md) to make sure my idea is not listed under the "A few things you shouldn't suggest" category.

**Comment by itcannotbe** [comment]
Isn't this, useless?

**Comment by NrkSed** [comment]
No because in late game when there are alot of enemies incoming you can get alot of power to compensate the power dip from your weapons

**Comment by papabreadd** [comment]
This sounds really interesting but the idea might need to be developed a little bit more in my opinion, mostly in terms of balancing changes and limitations

**Comment by itcannotbe** [comment]
> No because in late game when there are alot of enemies incoming you can get alot of power to compensate the power dip from your weapons

Most games have excess power available for their weapons already, also batteries.

**Comment by papabreadd** [comment]
Yeah the ability to spam batteries would be the most major issue I think

**Comment by OnyxFinger** [comment]
>  I also wonder if the opposite effect would be more interesting where energy consumption is proportional to damage

The very first iteration of the force projector worked that way (it would take power depending on the damage it takes, and never go down until you run out of power) but Anuke didn't like it.

**Comment by ChivalricMender** [comment]
Also called using a bunch of repair turrets on a single unit...
